### DISCUSSION TITLE
urllib3.exceptions.ProxyError:

### DISCUSSION THREAD

**Discussion by xuebaocpabl** [ricklamers/gpt-code-ui#38] [General]
I encountered the following problem. Please help me. Thank you.
urllib3.exceptions.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)

The above exception was the direct cause of the following exception:

urllib3.exceptions.ProxyError: ('Unable to connect to proxy. Your proxy appears to only use HTTP and not HTTPS, try changing your proxy URL to be HTTP. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#https-proxy-error-http-proxy', SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)')))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\python\lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
  File "D:\python\lib\site-packages\urllib3\connectionpool.py", line 844, in urlopen
    retries = retries.increment(
  File "D:\python\lib\site-packages\urllib3\util\retry.py", line 515, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Unable to connect to proxy. Your proxy appears to only use HTTP and not HTTPS, try changing your proxy URL to be HTTP. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#https-proxy-error-http-proxy', SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)'))))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\python\lib\site-packages\flask\app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
  File "D:\python\lib\site-packages\flask\app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "D:\python\lib\site-packages\flask_cors\extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "D:\python\lib\site-packages\flask\app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "D:\python\lib\site-packages\flask\app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "D:\python\lib\site-packages\gpt_code_ui\webapp\main.py", line 200, in generate_code
    code, status = loop.run_until_complete(
  File "D:\python\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "D:\python\lib\site-packages\gpt_code_ui\webapp\main.py", line 85, in get_code
    response = requests.post(
  File "D:\python\lib\site-packages\requests\api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
  File "D:\python\lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "D:\python\lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\python\lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "D:\python\lib\site-packages\requests\adapters.py", line 513, in send
    raise ProxyError(e, request=request)
requests.exceptions.ProxyError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Unable to connect to proxy. Your proxy appears to only use HTTP and not HTTPS, try changing your proxy URL to be HTTP. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#https-proxy-error-http-proxy', SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)'))))

**Comment by ricklamers** [comment]
https://stackoverflow.com/a/66874078

Your installed Python version might link against an outdated version of OpenSSL. Can you try installing the latest Python 3.10 build?

**Reply by xuebaocpabl** [reply]
> https://stackoverflow.com/a/66874078
> 
> Your installed Python version might link against an outdated version of OpenSSL. Can you try installing the latest Python 3.10 build?

This error message indicates that there is an issue connecting to the proxy server. It appears that the proxy server only supports HTTP and not HTTPS. This could be due to incorrect proxy configuration or the proxy server not supporting HTTPS connections. Is this the cause of the issue? Thank you for your response.

**Reply by xuebaocpabl** [reply]
Thanks for the reply. The version it prints out is OpenSSL 1.1.1k

**Reply by xuebaocpabl** [reply]
How can I resolve this? I really want to use your program. Thank you very much.

**Reply by xuebaocpabl** [reply]
According to the error message URL you mentioned, which I searched for, could you please explain how to modify the code? Is that approach feasible? here is :

If you’re receiving the [ProxyError](https://urllib3.readthedocs.io/en/latest/reference/urllib3.exceptions.html#urllib3.exceptions.ProxyError) and it mentions your proxy only speaks HTTP and not HTTPS here’s what to do to solve your issue:

If you’re using urllib3 directly, make sure the URL you’re passing into [urllib3.ProxyManager](https://urllib3.readthedocs.io/en/latest/reference/urllib3.poolmanager.html#urllib3.ProxyManager) starts with http:// instead of https://:

# Do this:
http = urllib3.ProxyManager("http://...")

# Not this:
http = urllib3.ProxyManager("https://...")

**Reply by ricklamers** [reply]
Can you try upgrading urllib3 to the latest version? `python -m pip install --upgrade urllib3`

**Reply by xuebaocpabl** [reply]
> Can you try upgrading urllib3 to the latest version? `python -m pip install --upgrade urllib3`

Requirement already satisfied: urllib3 in d:\python\lib\site-packages (2.0.3) ，Thanks for the reply

**Reply by xuebaocpabl** [reply]
urllib3.exceptions.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)

The above exception was the direct cause of the following exception:

urllib3.exceptions.ProxyError: ('Unable to connect to proxy. Your proxy appears to only use HTTP and not HTTPS, try changing your proxy URL to be HTTP. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#https-proxy-error-http-proxy', SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1129)')))     
I have searched for a lot of information, but I still can't resolve it.Thanks for the reply

**Reply by ricklamers** [reply]
Based on your stacktrace it looks like it’s just the Python `requests` package that’s calling urllib3. Have you tried a clean installation of Python?
